pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -0,0 +1,309 @@
+            state_indices=state_indices,
+        )
+
+        return f"{op_1_indices},...{state_indices}->...{new_state_indices}"
+
+    num_ch_wires = len(obs.wires)
+    einsum_indices = get_einsum_mapping(obs, state, map_indices, is_state_batched)
+    obs_mat = obs.matrix()
+    obs_shape = [QUDIT_DIM] * num_ch_wires * 2
+    obs_mat = math.cast(math.reshape(obs_mat, obs_shape), complex)
+    return math.einsum(einsum_indices, obs_mat, state)
+
+
+def trace_method(
+    measurementprocess: ExpectationMP, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Measure the expectation value of an observable by finding the trace of obs@rho.
+
+    Args:
+        measurementprocess (ExpectationMP): measurement process to apply to the state.
+        state (TensorLike): the state to measure.
+        is_state_batched (bool): whether the state is batched or not.
+
+    Returns:
+        TensorLike: the result of the measurement
+    """
+    obs = measurementprocess.obs
+    rho_mult_obs = apply_observable_einsum(obs, state, is_state_batched)
+
+    # using einsum since trace function axis selection parameter names
+    # are not consistent across interfaces, they don't exist for torch
+
+    num_wires = get_num_wires(state, is_state_batched)
+    trace = math.einsum(f"...{alphabet[:num_wires]*2}", rho_mult_obs)
+    return math.real(trace)
+
+
+def reduce_density_matrix(  # TODO: ask if I should have state diagonalization gates?
+    measurementprocess: StateMeasurement, state: TensorLike, is_state_batched: bool = False
+) -> TensorLike:
+    """Get the state or reduced density matrix.
+
+    Args:
+        measurementprocess (StateMeasurement): measurement to apply to the state
+        state (TensorLike): state to apply the measurement to
+        is_state_batched (bool): whether the state is batched or not
+
+    Returns:
+        TensorLike: the resulting resquared density_matrix
+    """
+    wires = measurementprocess.wires
+    if not wires:
+        return resquare_state(state, get_num_wires(state, is_state_batched))
+
+    num_state_wires = get_num_wires(state, is_state_batched)
+    wires_to_trace = [x + is_state_batched for x in range(num_state_wires) if x not in wires]
+
+    for i, wire_to_trace in enumerate(wires_to_trace):
+        axis1 = wire_to_trace - i
+        axis2 = axis1 + num_state_wires - i
+
+        print(axis1, axis2)
+
+        state = math.trace(state, axis1=axis1, axis2=axis2)
+
+    return resquare_state(state, len(wires))
+
+
+def _get_probs(state, num_wires, is_state_batched: bool = False):
+    """Finds the probabilities of measuring states in the computational basis
+
+    Args:
+        state (array[complex]): Input quantum state
+        num_wires (int): The number of wires the state represents
+        is_state_batched (bool): Boolean representing whether the state is batched or not
+
+    Returns:
+        Tensorlike: Vector representing probability of each state
```

Comment:
```suggestion
        state (array[complex]): input quantum state
        num_wires (int): the number of wires the state represents
        is_state_batched (bool): Boolean representing whether the state is batched or not

    Returns:
        Tensorlike: vector representing probability of each state
```
Minor formatting point for consistency.